opm/simulators/utils/ParallelRestart.hpp

```diff
@@ -361,6 +361,53 @@ ADD_PACK_PROTOTYPES(RestartValue)
 ADD_PACK_PROTOTYPES(std::string)
 ADD_PACK_PROTOTYPES(time_point)
 
+template<typename T, typename... Args>
+void variadic_packsize(size_t& size, Parallel::Communication comm, T& first, Args&... args)
+{
+  size += packSize(first, comm);
+  if constexpr (sizeof...(args) > 0)
+      variadic_packsize(size, comm, args...);
+}
```

Comment:
Any reason why you don't preserve the value categories of `Args` here?

Reply:
probably ignorance. i don't quite know what you are referring to.

Reply:
> i don't quite know what you are referring to.

Taking the pack by reference, instead of forwarding reference, and not `std::forward<>`-ing it on to the callee:
```C++
template <typename T, typename... Args>
void variadic_packsize(..., Args&&... args)
{
    // ...
    variadic_packsize(size, comm, std::forward<Args>(args)...)
}
```

Reply:
ah, i gotcha. that is of course better, no reason why i didn't, so changed.